cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorTest.java

```diff
@@ -1398,36 +1376,38 @@ public void testReconcile(Params params, VertxTestContext context) {
         ops.reconcileAll("test", kafkaNamespace, context.succeeding(v -> completeTest.flag()));
     }
 
+
     @ParameterizedTest
     @MethodSource("data")
     @Timeout(value = 2, timeUnit = TimeUnit.MINUTES)
-    public void testReconcileAllNamespaces(Params params, VertxTestContext context) {
+    @SuppressWarnings("unchecked")
+    public void testReconcileAllNamespaces(Params params, Vertx vertx, VertxTestContext context) {
         setFields(params);
 
         // create CM, Service, headless service, statefulset
         ResourceOperatorSupplier supplier = ResourceUtils.supplierWithMocks(openShift);
         ClusterOperatorConfig config = ResourceUtils.dummyClusterOperatorConfig(VERSIONS);
-        CrdOperator mockKafkaOps = supplier.kafkaOperator;
+        var mockKafkaOps = supplier.kafkaOperator;
         KafkaSetOperator mockKsOps = supplier.kafkaSetOperations;
         SecretOperator mockSecretOps = supplier.secretOperations;
+        String kafkaNamespace = "test";
 
         Kafka foo = getKafkaAssembly("foo");
-        foo.getMetadata().setNamespace("namespace1");
         Kafka bar = getKafkaAssembly("bar");
-        bar.getMetadata().setNamespace("namespace2");
-        when(mockKafkaOps.listAsync(eq("*"), any(Optional.class))).thenReturn(
+        when(mockKafkaOps.listAsync(eq(kafkaNamespace), any(Optional.class))).thenReturn(
                 Future.succeededFuture(asList(foo, bar))
         );
         // when requested Custom Resource for a specific Kafka cluster
-        when(mockKafkaOps.get(eq("namespace1"), eq("foo"))).thenReturn(foo);
-        when(mockKafkaOps.get(eq("namespace2"), eq("bar"))).thenReturn(bar);
-        when(mockKafkaOps.getAsync(eq("namespace1"), eq("foo"))).thenReturn(Future.succeededFuture(foo));
-        when(mockKafkaOps.getAsync(eq("namespace2"), eq("bar"))).thenReturn(Future.succeededFuture(bar));
+        when(mockKafkaOps.get(eq(kafkaNamespace), eq("foo"))).thenReturn(foo);
+        when(mockKafkaOps.get(eq(kafkaNamespace), eq("bar"))).thenReturn(bar);
+        when(mockKafkaOps.getAsync(eq(kafkaNamespace), eq("foo"))).thenReturn(Future.succeededFuture(foo));
+        when(mockKafkaOps.getAsync(eq(kafkaNamespace), eq("bar"))).thenReturn(Future.succeededFuture(bar));
```

Comment:
Sorry, this is still not good. The tests covers reconciliation over multiple namespaces. So I think it is important to use multiple namespaces.

Reply:
Okay, I think I see what I did with these tests, because I was rather confused about namespace changes that my commit logs showed I'd made, I somehow managed to swap the names or positions of them when rebasing... ...sweet merciful Jesus. 

I'll pull the original state for those two tests, and then sort it out from there. :/